types/video.js/index.d.ts

```diff
@@ -1986,9 +1986,10 @@ declare namespace videojs {
     };
 
     interface ComponentOptions {
-        children?: undefined | Child[];
-        createEl?: boolean;
-        el?: HTMLElement;
+        children?: Child[] | undefined;
+        createEl?: boolean | undefined;
+        el?: HTMLElement | undefined;
+        id?: string | undefined;
```

Comment:
given that this property is undocumented on video.js I don't believe it belongs in type definitions. they mention in the other issue (https://github.com/videojs/video.js/issues/7670) that this property SHOULD be documented but it isn't yet. my vote is that this does *not yet* belong here.